server/app/schemas.py

```diff
@@ -89,38 +157,127 @@ class Config:
 
 
 class Annotation(BaseModel):
+    """
+    Base Class defining an Annotation.
+    Annotations are considered as extra
+    information for a construct.
+
+    Args:
+        BaseModel (_type_): _description_
+    """
+
     key: str
     value: str
 
     class Config:
+        """Orm mode configuration"""
+
         orm_mode = True
 
 
-class Vector(BaseModel):
-    id: int = 0
+class VectorBase(BaseModel):
+    """
+    Base Class for defining a Vector.
+    The Vector class contains all fields that are
+    shared with all the other Vector classes,
+    except the VectorFromGenbank class.
+
+    Args:
+        BaseModel (_type_): _description_
+    """
+
     location: int
     name: str
+    bsa1_overhang: str
+    cloning_technique: str
     bacterial_strain: str
-    responsible: str
     group: str
-    bsa1_overhang: str
     selection: str
-    cloning_technique: str
+    responsible: str
     is_BsmB1_free: str
     notes: str
     REase_digest: str
+
+
+class VectorFromGenbank(BaseModel):
+    """
+    Base Class for defining a Vector.
+    This class contains all the vector information
+    that is read from a genbank file.
+
+    Args:
+        BaseModel (_type_): _description_
+    """
+
     sequence: str
     annotations: List[Annotation]
     features: List[Feature]
     references: List[Reference]
-    users: List[User]
-    level: Optional[VectorLevel]
-    BsmB1_site: Optional[str]
+
+
+class Vector(VectorBase):
+    """
+    This class inherits from the VectorBase class.
+    This contains additional information shared by
+    most Vector classes.
+
+    Args:
+        VectorBase (_type_): _description_
+    """
+
+    id: int = 0
     children: List["Vector"]
+    annotations: List[Annotation]
+    features: List[Feature]
+    references: List[Reference]
     bsmb1_overhang: str
+    users: List[User]
+    level: VectorLevel
     gateway_site: str
     vector_type: str
     date: Optional[datetime]
 
+
+class VectorInDB(Vector):
+    """
+    This class inherits from the Vector class.
+    It contains all the necessary information for
+    a Vector object when it is stored in the database.
+
+    Args:
+        Vector (_type_): _description_
+    """
+
+    sequence: str
+
     class Config:
+        """Orm mode configuration"""
+
         orm_mode = True
+
+
+class VectorToAdd(VectorBase):
+    """
+    This class inherit from the VectorBase class.
+    It contains supplemental information provided
+    when a new vector is posted to the database.
+
+    Args:
+        VectorBase (_type_): _description_
+    """
+
+    date: str
+    genbank_content: str
+
+
+class VectorOut(Vector):
+    """
+    This class inherit from the Vector class.
+    It contains all the necessary information for
+    a Vector object when it is queried from the database.
+
+    Args:
+        Vector (_type_): _description_
```

Comment:
```suggestion
    It contains all the necessary information for
    a Vector object when it is queried from the database.
    Not that the `sequence` is **NOT** sent, but instead
    is replaced by just the `sequence_length`. This is because
    the sequence is not used by the client but the length is
    so we save bytes over the wire.
```